File: web/src/main/java/by/sorface/passport/web/services/users/social/YandexOAuth2UserService.java
package by.sorface.passport.web.services.users.social;

import by.sorface.passport.web.dao.models.enums.ProviderType;
import by.sorface.passport.web.records.socialusers.YandexOAuth2User;
import by.sorface.passport.web.services.users.RoleService;
import by.sorface.passport.web.services.users.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class YandexOAuth2UserService extends AbstractSocialOAuth2UserService<YandexOAuth2User>
        implements SocialOAuth2UserService<YandexOAuth2User> {

    @Autowired
    protected YandexOAuth2UserService(final UserService userService,
                                      final RoleService roleService) {
        super(ProviderType.YANDEX, userService, roleService);
    }

}

File: backend/src/main/kotlin/by/sorface/idp/service/oauth/jdbc/YandexOAuth2UserService.kt
package by.sorface.idp.service.oauth.jdbc

import by.sorface.idp.config.security.oauth2.records.YandexOAuth2User
import by.sorface.idp.dao.sql.model.enums.ProviderType
import by.sorface.idp.dao.sql.repository.user.RoleRepository
import by.sorface.idp.dao.sql.repository.user.UserRepository
import by.sorface.idp.service.oauth.SocialOAuth2UserService
import org.springframework.stereotype.Service

@Service
class YandexOAuth2UserService(
    userRepository: UserRepository,
    roleRepository: RoleRepository
) : AbstractSocialOAuth2UserService<YandexOAuth2User>(ProviderType.YANDEX, userRepository, roleRepository), SocialOAuth2UserService<YandexOAuth2User>

